IdentityCore/src/cache/mac/MSIDMacCredentialStorageItem.m

```diff
@@ -201,4 +297,70 @@ - (NSPredicate *)createPredicateForKey:(MSIDCacheKey *)key
     return [NSCompoundPredicate andPredicateWithSubpredicates:subPredicates];
 }
 
+- (id<MSIDJsonSerializable, MSIDKeyGenerator>)getItemTypeFromString:(NSDictionary *)itemDict forKey:(NSString *)typeKey error:(NSError * __autoreleasing *)error
+{
+    if ([typeKey isEqualToString:MSID_ACCESS_TOKEN_CACHE_TYPE])
```

Comment:
nit: mappings dictionary could simplify some of this logic

Reply:
Leaving this as is for now as you will still need to load all these values to the dictionary first.